addons/mail/models/statusbar_time_mixin.py

```diff
@@ -0,0 +1,76 @@
+import logging
+from datetime import timedelta
+
+from odoo import fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class StatusbarTimeMixin(models.AbstractModel):
+    _name = "statusbar.time.mixin"
+    _description = "Mixin to compute the time a record spent in each stage"
+    _stage_field = "stage_id"
+
+    json_status_time = fields.Json(
+        string="Status time", compute="_compute_json_status_time", help="JSON that maps stage ids to seconds spent")
+
+    def _compute_json_status_time(self):
+        """
+        Computes json_status_time, a Json field stored as { <stage_id (str)>: <time_spent_in_seconds (int)> }
+
+            e.g. {"1": 1230, "3": 2220, "5": 14}
+        """
+
+        field = self.env['ir.model.fields'].sudo().search_fetch([
```

Comment:
Maybe we could re-use existing code and use ``_track_get_fields`` which returns the subset of tracked fields, like
```
if field not in self._track_get_fields():
    raise ValueError(
        _('...')
    )
```